ampd/src/sui/json_rpc.rs

```diff
@@ -22,18 +26,24 @@ where
 {
     async fn transaction_blocks(
         &self,
-        digests: Vec<TransactionDigest>,
-    ) -> Result<Vec<SuiTransactionBlockResponse>> {
+        digests: HashSet<TransactionDigest>,
+    ) -> Result<Vec<Option<SuiTransactionBlockResponse>>> {
         self.request(
             "sui_multiGetTransactionBlocks",
             (
                 digests
                     .iter()
-                    .map(|d| d.base58_encode())
+                    .map(TransactionDigest::base58_encode)
                     .collect::<Vec<_>>(),
                 SuiTransactionBlockResponseOptions::new().with_events(),
             ),
         )
         .await
+        .map(|vec: Vec<SuiTransactionBlockResponse>| {
+            vec.into_iter()
+                // None checkpoint means the transaction is not finalized
+                .map(|block| block.checkpoint.map(|_| block))
```

Comment:
Their docs mention `"This is only returned in the read api, not in the transaction execution api"`. What's the difference between the read API and the execution API? Also, it seems that this `checkpoint` is only Some(...) when the transaction is included as well as finalized. Can you rename the method `transaction_blocks` to `finalized_transaction_blocks` then?

Reply:
sure good name. the execution api execute transaction to blockchain and wait for response